api/logfilter/logfilter.go

```diff
@@ -134,11 +140,90 @@ func (l *LogFilter) ExistInBloomFilter(filter bloom.BloomFilter) bool {
 		}
 
 		for _, v := range e.Topic {
-			if filter.Exist(v) {
+			if bf.Exist(v) {
 				return true
 			}
 		}
 	}
 	// {} or nil matches any address or topic list
 	return len(l.pbFilter.Topics) == 0
 }
+
+// ExistInBloomFilterv2 returns true if addresses and topics of filter exist in the range bloom filter (topic: position-sensitive)
+func (l *LogFilter) ExistInBloomFilterv2(bf bloom.BloomFilter) bool {
+	if len(l.pbFilter.Address) > 0 {
+		flag := false
+		for _, addr := range l.pbFilter.Address {
+			if bf.Exist([]byte(addr)) {
+				flag = true
+			}
+		}
+		if !flag {
+			return false
+		}
+	}
+
+	if len(l.pbFilter.Topics) > 0 {
+		for i, e := range l.pbFilter.Topics {
+			if e == nil || len(e.Topic) == 0 {
+				continue
+			}
+			match := false
+			for _, v := range e.Topic {
+				if bf.Exist(append(byteutil.Uint64ToBytes(uint64(i)), v...)) {
+					match = true
+					continue
+				}
+			}
+			if !match {
+				return false
+			}
+		}
+		return true
+	}
+
+	// {} or nil matches any address or topic list
+	return true
+}
+
+// SelectBlocksFromRangeBloomFilter filters out RangeBloomFilter for selecting block numbers which have logFilter's topics/address
+// TODO[dorothy]: optimize using goroutine
+func (l *LogFilter) SelectBlocksFromRangeBloomFilter(bf bloom.BloomFilter, start, end uint64) []uint64 {
```

Comment:
This function is only height-sensitive. 

This is because for range level-bloomfilter and block level bloomfilter, we already check topic's position-sensitive filtering via ```ExistInBloomFilterv2```